HaaLimits/python/HaaLimits.py

```diff
@@ -71,138 +73,122 @@ def initializeWorkspace(self):
-            lamb = [-2,-5,0],
-        )
-        nameC4 = 'cont4{}'.format('_'+tag if tag else '')
-        cont4.build(self.workspace,nameC4)
     
-        #cont = Models.Sum('cont',
-        #    **{
-        #        nameC1 : [0.95,0,1],
-        #        nameC2 : [0.05,0,1],
-        #        'recursive' : True,
-        #    }
-        #)
-        #nameC = 'cont{}'.format('_'+tag if tag else '')
-        #cont.build(self.workspace,nameC)
+            # continuum background
+            cont = Models.Chebychev('cont',
+                order = 2,
+                p0 = [-1,-1.4,0],
+                p1 = [0.25,0,0.5],
+                p2 = [0.03,-1,1],
+            )
+            nameC = 'cont{}'.format('_'+tag if tag else '')
+            cont.build(self.workspace,nameC)
+        
+            cont1 = Models.Exponential('cont1', lamb = [-0.20,-1,0],  )
+            nameC1 = 'cont1{}'.format('_'+tag if tag else '')
+            cont1.build(self.workspace,nameC1)
     
-        # sum
-        bgs = {'recursive': True}
-        # continuum background
-        bgs[nameC1] = [0.5,0,1]
-        # jpsi
-        if self.XRANGE[0]<=4:
-            bgs[nameJ1] = [0.9,0,1]
-            bgs[nameJ2] = [0.9,0,1]
-            bgs[nameC3] = [0.5,0,1]
-        # upsilon
-        if self.XRANGE[0]<=9 and self.XRANGE[1]>=11:
-            bgs[nameU1] = [0.9,0,1]
-            bgs[nameU2] = [0.9,0,1]
-            bgs[nameU3] = [0.9,0,1]
-        bg = Models.Sum('bg', **bgs)
-        name = 'bg_{}'.format(region)
-        bg.build(self.workspace,name)
+            cont2 = Models.Exponential('cont2',
+                lamb = [-0.05,-1,0],
+            )
+            nameC2 = 'cont2{}'.format('_'+tag if tag else '')
+            cont2.build(self.workspace,nameC2)
+        
+            cont3 = Models.Exponential('cont3',
+                lamb = [-0.75,-5,0],
+            )
+            nameC3 = 'cont3{}'.format('_'+tag if tag else '')
+            cont3.build(self.workspace,nameC3)
+        
+            cont4 = Models.Exponential('cont4',
+                lamb = [-2,-5,0],
+            )
+            nameC4 = 'cont4{}'.format('_'+tag if tag else '')
+            cont4.build(self.workspace,nameC4)
 
+            # sum
+            bgs = {'recursive': True}
+            # continuum background
+            bgs[nameC1] = [0.5,0,1]
+            # jpsi
+            if self.XRANGE[0]<=4:
+                bgs[nameJ1] = [0.9,0,1]
+                bgs[nameJ2] = [0.9,0,1]
+                bgs[nameC3] = [0.5,0,1]
+            # upsilon
+            if addUpsilon:  #self.XRANGE[0]<=9 and self.XRANGE[1]>=11:
+        	print "ADDING UPSILON"
+                bgs[nameU1] = [0.9,0,1]
+                bgs[nameU2] = [0.9,0,1]
+                bgs[nameU3] = [0.9,0,1]
+                bg = Models.Sum('bg', **bgs)
+                name = 'bg_{}'.format(region)
+            else:
+                bg = Models.Sum('bg', **bgs)
+	        name = 'bg_sides_{}'.format(region)
```

Comment:
Why the choice for two names?
Keep the same name, the workspace can be recreate/overwritten when you change later